test/interval_tests/numeric.jl

```diff
@@ -434,3 +434,43 @@ end
     @test nthroot(Interval{BigFloat}(-81, -16), -4) == ∅
     @test nthroot(Interval{BigFloat}(-81, -16), 1) == Interval{BigFloat}(-81, -16)
 end
+
+# approximation used for testing (not to rely on ≈ for intervals)
+# ⪆(x, y) = (x ≈ y) && (x ⊇ y)
+⪆(x::Interval, y::Interval) = x.lo ≈ y.lo && x.hi ≈ y.hi && x ⊇ y
+
+@testset "`mod`" begin
+    r = 0.0625
+    x = r..(1+r)
+    @test mod(x, 1) == mod(x, 1.0) == 0..1
+    @test mod(x, 2) == mod(x, 2.0) ⪆ x
+    @test mod(x, 2.5) ⪆ x
+    @test mod(x, 0.5) == 0..0.5
+    @test mod(x, -1) == mod(x, -1.0) == -1..0
+    @test mod(x, -2) == mod(x, -2.0) ⪆ -2+x
+    @test mod(x, -2.5) ⪆ -2.5+x
+    @test mod(x, -0.5) == -0.5..0
+
+    x = (-1+r) .. -r
+    @test mod(x, 1) == mod(x, 1.0) ⪆ 1+x
+    @test mod(x, 2) == mod(x, 2.0) ⪆ 2+x
+    @test mod(x, 2.5) ⪆ 2.5+x
+    @test mod(x, 0.5) == 0..0.5
+    @test mod(x, -1) == mod(x, -1.0) ⪆ x
+    @test mod(x, -2) == mod(x, -2.0) ⪆ x
+    @test mod(x, -2.5) ⪆ x
+    @test mod(x, -0.5) == -0.5..0
+
+    x = -r .. 1-r
+    @test mod(x, 1) == mod(x, 1.0) == 0..1
+    @test mod(x, 2) == mod(x, 2.0) == 0..2
+    @test mod(x, 2.5) == 0..2.5
+    @test mod(x, 0.5) == 0..0.5
+    @test mod(x, -1) == mod(x, -1.0) == -1..0
+    @test mod(x, -2) == mod(x, -2.0) == -2..0
+    @test mod(x, -2.5) == -2.5..0
+    @test mod(x, -0.5) == -0.5..0
+
+    # TODO - implement mod for two intervals
+    @test_throws TypeError mod(1..2, 1.4..1.5)
```

Comment:
this is causing problems on nightly. I think it's better not to rely on what exception the julia internals throw, as they may change (as this example shows), see [this comment](https://github.com/JuliaIntervals/IntervalArithmetic.jl/pull/525#discussion_r883348123) .

Reply:
Solved by defining `ArgumentError` explicitly, as you proposed.